clients/src/main/java/org/apache/kafka/common/security/authenticator/SaslClientAuthenticator.java

```diff
@@ -326,6 +358,11 @@ public Long reauthenticationLatencyMs() {
         return reauthInfo.reauthenticationLatencyMs();
     }
 
+    private int nextCorrelationId() {
+        if (!isReserved(correlationId)) correlationId = MIN_RESERVED_CORRELATION_ID;
```

Comment:
as before, this could be if-then-else without the increment below for the `if` case.